File: backend/fix_tripay.js
const fs = require('fs');
let content = fs.readFileSync('src/integrations/providers/payment/tripay/tripay.module.ts', 'utf8');
if (!content.includes('GlModule')) {
    content = "import { GlModule } from '../../../../../gl/gl.module';\n" + content;
    content = content.replace("imports: [PrismaModule, PaymentModule]", "imports: [PrismaModule, PaymentModule, GlModule]");
    fs.writeFileSync('src/integrations/providers/payment/tripay/tripay.module.ts', content);
}

File: backend/fix_tripay.py
with open('src/integrations/providers/payment/tripay/tripay.controller.ts', 'r') as f:
    c = f.read()

c = c.replace("external_type: 'TRIPAY_TRANSACTION'", "entity_type: 'TRIPAY_TRANSACTION'")
c = c.replace(", orderBy: { created_at: 'desc' }", "")

with open('src/integrations/providers/payment/tripay/tripay.controller.ts', 'w') as f:
    f.write(c)
